pdns/pdns_recursor.cc

```diff
@@ -1778,7 +1780,7 @@ static void startDoResolve(void *p)
         else {
           Utility::gettimeofday(&g_now, 0); // needs to be updated
           struct timeval ttd = g_now;
-          if (dc->d_tcpConnection->d_requestsInFlight == dc->d_tcpConnection->d_maxInFlight - 1) {
+          if (dc->d_tcpConnection->d_requestsInFlight == TCPConnection::s_maxInFlight - 1) {
```

Comment:
this would be better formed as `dc->d_tcpConnection->d_requestsInFlight + 1 == TCPConnection::s_maxInFlight` given it's unsigned int.

Reply:
`TCPConnection::s_maxInFlight` should be `>= 1` (though we don't check for that atm).